cognite/powerops/client/shop/data_classes/shop_case.py

```diff
@@ -55,30 +56,40 @@ def __init__(
     ) -> None:
         if data and file_path:
             raise ValueError("Cannot specify both data and file_path")
-        self.data = {}
-        self.excess_yaml_parts: list[str] = []
+
         if data:
-            self.load_case_data(data)
-        elif file_path:
-            self.load_case_file(file_path)
+            self._case_string = data.lstrip("\n")
+        else:
+            self._case_string = self.load_case_file(file_path)
+
+        self._data: Union[dict, None] = None
+        self.excess_yaml_parts: list[str] = []
+
         self._shop_files: list[SHOPFileReference] = list(shop_files)
         self.watercourse = watercourse
         self.shop_version = shop_version
 
-    def load_case_data(self, data: str) -> None:
-        self.data = {}
+    def load_case_data(self) -> None:
         self.excess_yaml_parts = []
         try:
-            if yaml_docs := list(yaml.safe_load_all(data)):
-                self.data = yaml_docs[0]
+            if yaml_docs := list(yaml.safe_load_all(self._case_string)):
+                self._data = yaml_docs[0]
                 self._handle_excess_yaml_parts(yaml_docs[1:])
         except yaml.YAMLError as e:
             raise ValueError("Could not parse case data") from e
 
-    def load_case_file(self, file_path: str, encoding: str = "utf-8") -> Self:
+    @staticmethod
+    def load_case_file(file_path: str, encoding: str = "utf-8") -> str:
         with Path(file_path).open(encoding=encoding) as file:
-            self.load_case_data(file.read())
-        return self
+            return file.read()
+
+    @property
+    def data(self) -> dict:
+        if self._data is None:
+            self.load_case_data()
+
+        assert self._data is not None
```

Comment:
Should we perhaps use a "normal" if check rather than an assert?

Reply:
Think in assert isn't super abnormal, but can just initiliase this is an empty dict, as this is falsy and eliminate this code.